junipervpn/vpn.py

```diff
@@ -350,45 +351,55 @@ def main():
     parser.add_argument('--help', action='help',
                         help="Show help")
     args = parser.parse_args()
-    args.__dict__['password'] = None
-
-    if len(args.action) and args.action[0] == '--':
-        args.action = args.action[1:]
-
-    if not len(args.action):
-        args.action = None
-
-    logging.basicConfig(level=logging.DEBUG if args.verbose else logging.INFO)
 
+    # Load the conf file and use it as defaults, so it can be overidden by the
+    # command line
     if args.config is not None:
         config = configparser.RawConfigParser()
         config.read(args.config)
-        for arg in ['username', 'host', 'password', 'oath', 'action', 'stdin',
-                    'hostname', 'platform', 'hwaddr', 'certs', 'device_id',
-                    'user_agent', 'pass_prefix', 'realm']:
-            if args.__dict__[arg] is None:
-                try:
-                    args.__dict__[arg] = config.get('vpn', arg)
-                except configparser.NoOptionError:
-                    pass
-
-        if not args.enable_funk:
-            try:
-                val = config.get('vpn', 'enable_funk').lower()
-            except configparser.NoOptionError:
-                pass
-            else:
-                if val in ['true', '1', 'yes', 'enable', 'on']:
-                    args.enable_funk = True
-                elif val in ['false', '0', 'no', 'disable', 'off']:
-                    args.enable_funk = False
+        vpn_config = config['vpn']
+
+        identity = lambda x: x
+        conf_handlers = defaultdict(
+            # default
+            lambda: (vpn_config.get, identity),
+
+            action=(vpn_config.get, shlex.split),
+            verbose=(vpn_config.getboolean, identity),
+            enable_funk=(vpn_config.getboolean, identity),
+        )
+
+        def parse(arg):
+            handlers = conf_handlers[arg]
```

Comment:
Would it work to do something like:

```
get_fn, process_fn = config_handlers[arg]
```

and use them instead of `handlers[1]` and `handlers[0]`? That might be a bit easier to read.

(or better names if you can think of them!)

Reply:
Went for 
```
            from_conf, post_process = conf_handlers[arg]
```